src/main/scala/com/metamx/tranquility/beam/ClusteredBeam.scala

```diff
@@ -320,9 +386,80 @@ class ClusteredBeam[EventType: Timestamper, InnerBeamType <: Beam[EventType]](
     }
   }
 
+  def groupEvents(eventType: EventType, clusteredBeamMeta: ClusteredBeamMeta): DateTime = {
+    allowGranularityChange = false
+    val eventTimestamp = timestamper(eventType)
+    val eventBucket = tuning.segmentBucket(eventTimestamp)
+    // ZK does not know about any beams
```

Comment:
yes it is describing the `if` I will make it more clear and move it inside the block